scripts/tasks/api-extractor.ts

```diff
@@ -14,10 +16,79 @@ export function apiExtractor() {
   return apiExtractorConfigs.length
     ? series(
         ...apiExtractorConfigs.map(([configPath, configName]) => {
+          // note we have `"strictNullChecks": false,` turned off - so from TS point of view this always returns API 🚨
+          const overrideApi = overrideExtractorConfigForPackagesWithTsPathAliases({
+            apiExtractorConfigPath: configPath,
+            tsConfigPath: resolveCwd('./tsconfig.json'),
+          });
+
           const taskName = `api-extractor:${configName}`;
-          task(taskName, apiExtractorVerifyTask({ configJsonFilePath: configPath, localBuild }));
+
+          task(
+            taskName,
+            series(
+              () => {
+                overrideApi && overrideApi.overrideConfig();
+              },
+              apiExtractorVerifyTask({ configJsonFilePath: configPath, localBuild }),
+              () => {
+                overrideApi && overrideApi.resetConfig();
```

Comment:
I think doing it like this is going to leave the API Extractor config in a modified state if `apiExtractorVerifyTask` fails.

Reply:
true that. I think this shouldn't be a problem on CI ?  as if this fails we will bail the pipeline

Reply:
https://github.com/microsoft/fluentui/pull/19040/commits/b2883e2414c5f3c533dc9e71ea5b8c36ebc0431a

Reply:
Thanks for the update! My concern was more about the local build scenario, since leaving the files in a bad state could confuse people (and maybe let the changes accidentally get checked in).

Reply:
that's indeed a good concern and I appreciate pointing that out 🙌.

btw dunno why but after I added that try/catch pipeline started to fail on api-docs generate-json task 🧐...

Reply:
> btw dunno why but after I added that try/catch pipeline started to fail on api-docs generate-json task 🧐...

ok looks like I figured it out - turns out just scripts are not that intuitive as one would expect... my bad 👀